content/blog/2020-09-17-delivering-learnr-tutorials-in-a-package/index.Rmarkdown

```diff
@@ -0,0 +1,353 @@
+
+    * This creates an `inst/` directory with some subfolders. If you're not new to creating learnr tutorials, the folder structure here will look familiar to you -- there's nothing different about learnr tutorial directories inside and outside of packages.
+
+      You can open up `inst/` and its subsequent subfolders until you reach the `.Rmd` file you created:
+
+      ```
+      /inst/tutorials/<tutorial-name>/<tutorial-name.Rmd>
+      ```
+
+    ```{r, eval = TRUE, echo = FALSE}
+    knitr::include_graphics("pkg-inst.png")
+    ```
+
+1. **Edit the open `.Rmd` file** and write your tutorial as you normally would.
+
+1. **Repeat** for each additional tutorial you want to include in this package.
+
+
+
+### Build, install, and use your tutorial package
+
+Time to build the package and install it in your R system library.
+
+1. __Navigate to the *Build* pane__ of your RStudio IDE. Then __click *Install and Restart*__. 
+
+    * Alternatively, run:
+
+      ```{r, eval = FALSE}
+      devtools::install()
+      ```
+
+    * Believe it or not, you've now made a package with a learnr tutorial inside of it!
+
+    * You should test that your package has been properly created by loading your tutorial from your installed package:
+    
+1. __Navigate to the *Tutorial* pane__ of your RStudio IDE. All installed learnr tutorials in your R Library will be automatically indexed and displayed here. 
+
+    ```{r, echo = FALSE, out.width="90%"}
+    knitr::include_graphics("pkg-run-tutorial.png")
+    ```
+    
+    
+1. __Click Start Tutorial__ to open the tutorial from inside the tutorial pane (which can then be made bigger or popped out to a browser window) -- `r emo::ji("tada")` woot!
+  
+
+    * Alternatively, you can run your tutorial by using `learnr::run_tutorial("<tutorial-name>", "<package-name>")`.
+
+        ```{r, eval = FALSE}
+        learnr::run_tutorial("lesson1", "tutorialpackage")
+        ```
+
+    * The tutorial name here refers to the name of the `.Rmd` file, not the user-facing title.
+    * The live tutorial will open up in a local browser window.
+    
+
+### (Optionally) Add dependencies
+
+If your tutorial uses external packages (for example, packages for exercise-checking like [gradethis](https://rstudio-education.github.io/gradethis/)), you'll need to make sure this package gets installed along with your custom package. If the package you want to use is not on CRAN (as gradethis currently is not), it will need to be included as a `Remotes` dependency in the `DESCRIPTION` file. (*Note:* Including external packages under `Remotes`, as opposed to `Imports`, does complicate matters if you plan on submitting your tutorial package to CRAN later.)
+
+Once again, the usethis package will make this easier for us to do:
+
+
+1. **Run** `usethis::use_dev_package("<external-package>")`
+
+    ```{r, eval = FALSE}
+    usethis::use_dev_package("gradethis")
```

Comment:
See above comment on this, doesn't one need to specify the remote, or is `use_dev_package()` smart enough to know it? That seems unlikely it would know the `OWNER` of the repo?

Reply:
@mine-cetinkaya-rundel Just tested this, and I was surprised to see that it is smart enough to know the `OWNER` of the repo. Seems to be the case if the dev package I'm specifying is one that I have installed locally from a supported remote.  I'll make a little note of this in the post:

```
> usethis::use_dev_package("gradethis")
✓ Adding 'gradethis' to Imports field in DESCRIPTION
✓ Adding 'rstudio-education/gradethis' to Remotes field in DESCRIPTION
● Refer to functions with `gradethis::fun()`
> 

```

Stranger yet, I *cannot* seem to get `use_dev_package()` to accept the `remote` argument 😩